src/walk.rs

```diff
@@ -44,6 +46,40 @@ pub enum WorkerResult {
     Error(ignore::Error),
 }
 
+/// Converts a WorkerResult to a PathBuf. This is required as in some cases,
+/// the underlying `ignore` libary will return an error when we actually
+/// still want to show the path.
+///
+/// Currently, the cases supported are:
+///   - a broken symlink should still be found and the path returned
+impl TryFrom<WorkerResult> for PathBuf {
+    type Error = ignore::Error;
+
+    fn try_from(worker_result: WorkerResult) -> Result<PathBuf, Self::Error> {
+        match worker_result {
+            WorkerResult::Entry(value) => return Ok(value),
+            WorkerResult::Error(err) => {
+                if let ignore::Error::WithPath {
```

Comment:
Could this `if let` pattern matching be done in the outer `match` statement? Something like:
``` rust

        match worker_result {
            WorkerResult::Entry(value) => return Ok(value),
            WorkerResult::Error(ignore::Error::WithPath { path, err: path_err }) => …
            …
```

(maybe needs a `ref`)

This way, we could probably also get rid of the intermediate `return`s and just have one big expression.